build/flutter_app.gni

```diff
@@ -105,7 +104,7 @@ template("flutter_app") {
       "--output-file",
       rebase_path(bundle_path),
       "--snapshot",
-      rebase_path(snapshot_path),
+      rebase_path(snapshot_path, root_build_dir),
```

Comment:
This is related to how Dart generates depfiles. If the path to snapshot is absolute, Dart generates a depfile with an absolute path in the rule, similarly for relative paths. When using depfiles with Ninja, it expect paths in depfiles to be relative to the build directory.

Just for reference, I found CL [1601863006](https://codereview.chromium.org/1601863006/) which applied the same fix to Dart action used by Mojo.